src/app/components/wrapper-components/online-game-wrapper/OGWCTimeManagerService.ts

```diff
@@ -0,0 +1,171 @@
+    private allClocks: CountDownComponent[]; // Initialized by setClocks
+    /*
+     * The configRoom, which is set when starting the clocks.
+     * We need it to know the maximal game and move durations.
+     */
+    private configRoom: MGPOptional<ConfigRoom> = MGPOptional.empty();
+    // The global time taken by each player since the beginning of the part
+    private readonly takenGlobalTime: [number, number] = [0, 0];
+    // The global time added to each player
+    private readonly extraGlobalTime: [number, number] = [0, 0];
+    // The turn time available for each player. Distinct from the clocks so it stays constant within a turn
+    private readonly availableTurnTime: [number, number] = [0, 0];
+    // The time at which the current move started
+    private lastMoveStartTimestamp: MGPOptional<Timestamp> = MGPOptional.empty();
+
+    public setClocks(turnClocks: [CountDownComponent, CountDownComponent],
+                     globalClocks: [CountDownComponent, CountDownComponent])
+    : void
+    {
+        this.turnClocks = turnClocks;
+        this.globalClocks = globalClocks;
+        this.allClocks = turnClocks.concat(globalClocks);
+    }
+    // At the beginning of a game, set up clocks and remember when the game started
+    public onGameStart(configRoom: ConfigRoom): void {
+        this.configRoom = MGPOptional.of(configRoom);
+        for (const player of Player.PLAYERS) {
+            // We need to initialize the service's data
+            // Otherwise if we go to another page and come back, the service stays alive and the data is off
+            this.takenGlobalTime[player.value] = 0;
+            this.extraGlobalTime[player.value] = 0;
+            this.availableTurnTime[player.value] = this.getMoveDurationInMs();
+            // And we setup the clocks
+            this.globalClocks[player.value].setDuration(this.getPartDurationInMs());
+            this.turnClocks[player.value].setDuration(this.getMoveDurationInMs());
+        }
+        // We want the clocks to be paused, as we will only activate the required ones
+        for (const clock of this.allClocks) {
+            clock.start();
+            clock.pause();
+        }
+    }
+    private getPartDurationInMs(): number {
+        return this.configRoom.get().totalPartDuration * 1000;
+    }
+    private getMoveDurationInMs(): number {
+        return this.configRoom.get().maximalMoveDuration * 1000;
+    }
+    public onReceivedAction(action: PartEventAction): void {
+        switch (action.action) {
+            case 'AddTurnTime':
+                this.addTurnTime(Player.of(action.player));
+                break;
+            case 'AddGlobalTime':
+                this.addGlobalTime(Player.of(action.player));
+                break;
+            case 'StartGame':
+                this.lastMoveStartTimestamp = MGPOptional.of(action.time as Timestamp);
+                break;
+            case 'EndGame':
+                break;
```

Comment:
soit ça doit faire qqch soit il manque un commentaire là

Reply:
appel à this.endGame rajouté ici plutôt que de l'appeler depuis OGWC